bokehjs/src/lib/core/has_props.ts

```diff
@@ -394,13 +396,14 @@ export abstract class HasProps extends Signalable() implements Equals, Printable
       previous.set(prop, prop.get_value())
     }
 
-    this._setv(changed, options)
+    const updated = this._setv(changed, options)
 
     const {document} = this
     if (document != null) {
       const changed: [Property, unknown, unknown][] = []
       for (const [prop, value] of previous) {
-        changed.push([prop, value, prop.get_value()])
+        if (updated.has(prop))
+          changed.push([prop, value, prop.get_value()])
```

Comment:
Why can't this just loop over `updated` directly, rather than loop over `previous` and check `updated` each iteration?

Reply:
This definitively can be simplified. There was too much shuffling there even before this change.